File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs = sorted(costs)
        iceCream = i = 0
        while i < len(costs) and coins >= costs[i]:
            coins -= costs[i]
            i, iceCream = i + 1, iceCream + 1
        return iceCream

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(begin(costs), end(costs));
        int iceCream = 0, i = 0;
        while(i < size(costs) and coins >= costs[i]) {
            coins -= costs[i];
            ++i;
            ++iceCream;
        }
        return iceCream;
    }
};